base/collection_n1ql.go

```diff
@@ -119,8 +122,44 @@ func (c *Collection) CreatePrimaryIndex(indexName string, options *N1qlIndexOpti
 	return CreatePrimaryIndex(c, indexName, options)
 }
 
-func (c *Collection) WaitForIndexOnline(indexName string) error {
-	return WaitForIndexOnline(c, indexName)
+// WaitForIndexOnline takes set of indexes and watches them till they're online.
+func (c *Collection) WaitForIndexOnline(indexNames []string, watchPrimary bool) error {
```

Comment:
I think you can change this name to be `WaitForIndexes` or `WaitForIndexesOnline` and just have it take a slice of strings. We can look in the slice of strings to see if `PrimaryIndexName` is in it.

Reply:
In fact we don't have to do this, because gocb does this for us:

https://github.com/couchbase/gocb/blob/v2.6.0/cluster_queryindexes.go#L727